client.go

```diff
@@ -17,6 +17,8 @@ var ProcessExpiredAdInterval = 1 * time.Minute
 
 type CacheFunc func(string, time.Duration) (string, error)
 type EventFunc func(string, string, string, string)
+type PopHandler func(string, string, *ProofOfPlayRequest) (
```

Comment:
PoPFunc might be more correct - handler makes me think you are handling the result of a pop.